sound/soc/sof/sof-audio.h

```diff
@@ -46,6 +46,8 @@
 #define SOF_DAI_CLK_INTEL_SSP_MCLK	0
 #define SOF_DAI_CLK_INTEL_SSP_BCLK	1
 
+#define SOF_TRIGGER_MAX_PIPELINE_COUNT 16
```

Comment:
What the limitation of ```16``` comes from?

Reply:
thats the max number of pipelines in the FW

Reply:
The restriction is on the number of pipelines that have a window for timestamping. We have no fixed restriction to 16 in general, and if you see the number of DMAs we are past 16 already.

@bardliao added something to read the actual number of pipelines from the firmware, can you refresh everyone's memory?

Reply:
@plbossart that's correct but I need to create a fixed-size array for IPC playload here and I need a number. So I pixked 16 because it seemed like a large enough number for connected pipelines

Reply:
why not allocate dynamically based on what the firmware actually supports?

Reply:
@plbossart this is exactly how I started did this PR first and remember I told you the multi-pipeline IPC always fails with both the cavs and the SOF firmware? this is because I couldnt use dynamically allocated data with the IPC

Reply:
erm, what? we allocate such memory all the time. Not getting the issue, sorry.

Reply:
@plbossart could you please point me to an IPC structure containing a dynamically allocated data pointer?

Reply:
you can allocate a u8 buffer and cast to a structure as needed?

Reply:
isn't that what the 'count' is for, to tell the firmware how many pipelines to look for?

Reply:
but the count is not fixed isnt it? It depends on how many pipelines that PCM has

Reply:
let me rework the PR with a flex array. Maybe then I can removethe max count restriction

Reply:
> The restriction is on the number of pipelines that have a window for timestamping. We have no fixed restriction to 16 in general, and if you see the number of DMAs we are past 16 already.
> 
> @bardliao added something to read the actual number of pipelines from the firmware, can you refresh everyone's memory?

https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/ipc4-loader.c#L386 in `sof_ipc4_query_fw_configuration`. But I think @ranj063 already know it. :)